python/pyarrow/tests/test_table.py

```diff
@@ -2422,3 +2422,27 @@ def test_numpy_asarray(constructor):
     result = np.asarray(table3, dtype="int32")
     np.testing.assert_allclose(result, expected)
     assert result.dtype == "int32"
+
+
+def test_invalid_non_join_column():
+    NUM_ITEMS = 30
+    t1 = pa.Table.from_pydict({
+        'id': [x.to_bytes(4, 'big') for x in range(NUM_ITEMS)],
+        'array_column': [[z for z in range(3)] for x in range(NUM_ITEMS)],
+    })
+    t2 = pa.Table.from_pydict({
+        'id': [x.to_bytes(4, 'big') for x in range(NUM_ITEMS)],
```

Comment:
```suggestion
        'id': range(NUM_ITEMS),
```

Same comment as above.